cmd/test-server/zerotrace.go

```diff
@@ -0,0 +1,295 @@
+package main
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcap"
+	"golang.org/x/net/ipv4"
+)
+
+const (
+	beginTTLValue        = 5
+	MaxTTLHops           = 32
+	stringToSend         = "test string tcp"
+	tracerouteHopTimeout = time.Second * 10
+	snaplen              = 65536
+	promisc              = true
+	ipversion            = 4
+)
+
+var (
+	buffer  gopacket.SerializeBuffer
+	options = gopacket.SerializeOptions{
+		ComputeChecksums: true,
+		FixLengths:       true,
+	}
+	deviceName   string
+	icmpPktError = errors.New("IP header unavailable")
+)
+
+type SentPacketData struct {
+	HopIPId     uint16
+	HopSentTime time.Time
+}
+
+type HopRTT struct {
+	IP  net.IP
+	RTT float64
+}
+
+type zeroTrace struct {
+	Iface            string
+	Conn             net.Conn
+	UUID             string
+	PcapHdl          *pcap.Handle
+	ClientIP         string
+	ClientPort       int
+	IPIdHop          map[int][]SentPacketData
```

Comment:
Are we anticipating concurrent reads/writes to this map?

Reply:
(I've renamed this map to `SentPktsIPId`) 
It is a map of the IPIds and sent times of the sent packets for each TTL. I don't expect concurrent reads or writes because each packet---sent packet (writes to this) or received packet (reads from this)---is dealt with sequentially.